doc/conf.py

```diff
@@ -169,7 +169,11 @@
         "API": f"{reference_prefix}/doxygen/html/index.html",
         "Kconfig Options": f"{reference_prefix}/kconfig.html",
         "Devicetree Bindings": f"{reference_prefix}/build/dts/api/bindings.html",
-    }
+    },
+    # Set google_searchengine_id to your Search Engine ID to replace built-in search
+    # engine with Google's Programmable Search Engine.
+    # See https://programmablesearchengine.google.com/ for details.
+    "google_searchengine_id": "746031aa0d56d4912"
```

Comment:
Please check out latest update to the PR -- folks can now decide in the UI which engine they want to use. I guess we can still provide some level of build-time configuration though so that search engine selection widget is available only in case folks have enabled the use of a 3rd party engine when generating the docs.